app/components/graph-editor/make-fn-node.ts

```diff
@@ -0,0 +1,222 @@
+// File: make-fn-node.ts
+// Created by: CJ Dimaano
+// Date created: January 13, 2016
+//
+// THIS FILE IS INTENDED TO BE IMPORTED ONLY INTO graph-editor.component.ts
+//
+
+
+import * as CONST from "./constants";
+import * as canvas from "./canvas";
+import { DrawableNode } from "./drawable-interfaces";
+
+
+/**
+ * makeFnNode  
+ */
+export function makeFnNode(
```

Comment:
This seems like a fairly complex bit of code and should maybe have some more comments.